rust/serving/src/app/store/cbstore/jetstreamstore.rs

```diff
@@ -171,42 +172,95 @@ impl super::CallbackStore for JetStreamCallbackStore {
-                if entry.key == callbacks_init_key {
-                    continue;
-                }
+                    // done key is used to signal the end of processing, we can break the loop
+                    if entry.key == callbacks_done_key {
+                        tracing::info!(
+                            callbacks_done_key,
+                            "Received event for the callback_done_key. Stopping the watcher task"
+                        );
+                        break;
+                    }
 
-                // done key is used to signal the end of processing, we can break the loop
-                if entry.key == callbacks_done_key {
-                    break;
+                    let cbr: Callback = entry
+                        .value
+                        .try_into()
+                        .inspect_err(|err| tracing::error!(?err, "Failed to deserialize callback"))
+                        .expect("Failed to deserialize callback");
+
+                    tx.send(Arc::new(cbr))
+                        .await
+                        .inspect_err(|err| {
+                            tracing::error!(
+                                ?err,
+                                "Failed to send callback details from watcher task to main task"
+                            )
+                        })
+                        .expect("Failed to send callback");
                 }
-
-                let cbr: Callback = entry
-                    .value
-                    .try_into()
-                    .expect("Failed to deserialize callback");
-
-                tx.send(Arc::new(cbr))
-                    .await
-                    .expect("Failed to send callback");
+                if !received_events && attempts < 5 {
+                    tracing::warn!(
+                        callbacks_key,
+                        "Watcher for Jetstream key didn't return any events. Will recreate the watcher in 20ms"
+                    );
+                    tokio::time::sleep(Duration::from_millis(20)).await;
+                    watcher = kv_store
+                        .watch_from_revision(&callbacks_key, 1)
+                        .await
+                        .map_err(|e| {
+                            StoreError::StoreRead(format!(
+                                "Failed to watch request id in kv store: {e:?}"
+                            ))
+                        })
+                        .inspect_err(|err| {
+                            tracing::error!(?err, callbacks_key, "Failed to recreate the watcher")
+                        })
+                        .expect("Failed to recreate the watcher");
+                }
+            }
+            if !received_events {
+                tracing::error!(
+                    callbacks_key,
+                    "Watcher for Jetstream key didn't return any events even after retries"
+                );
             }
```

Comment:
what will happen because of this? exit the container and do restart?

Reply:
Currently, only that request will fail (500 status code). It won't result in container restart.